client.go

```diff
@@ -103,6 +105,36 @@ func NewClient(endpoint, apiKey, apiSecret string) *Client {
 	return client
 }
 
+// SetTLSAuthentication update the HTTP client with the provided user certificate/key
+func (client *Client) SetTLSAuthentication(caCertFile, certFile, keyFile string) error {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return err
+	}
+
+	caCert, err := ioutil.ReadFile(caCertFile)
+	if err != nil {
+		return err
+	}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{cert},
+		RootCAs:      caCertPool,
+	}
+	tlsConfig.BuildNameToCertificate()
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{TLSClientConfig: tlsConfig},
```

Comment:
By proceeding this way you overwrite a potentially configured Transport – e.g. if the user already has called [`Client.TraceOn()`](https://godoc.org/github.com/exoscale/egoscale#Client.TraceOn). You should instead call [`http.Transport.Clone()`](https://godoc.org/net/http#Transport.Clone) and then set the `TLSClientConfig` on the cloned transport, something like:

```go
tlsTransport := client.HTTPClient.Transport.Clone()
tlsTransport.TLSClientConfig = tlsConfig
client.HTTPClient = tlsTransport
```

(haven't tested it though 😉)